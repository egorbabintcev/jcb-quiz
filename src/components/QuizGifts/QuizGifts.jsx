import React from 'react';
import styles from './QuizGifts.module.scss';
import prices from './prices.png';
import gloves from './gloves.png';
import chat from './chat.png';

const QuizGifts = () => (
  <div className={styles.gifts}>
    <h4 className={styles.title}>Гарантированные подарки:</h4>
    <div className={styles.gifts_wrapper}>
      <div className={styles.gift}>
        <img src={prices} alt="" />
        <p className={styles.title}>
          <strong>3 варианта цены</strong>
          <br /> на оригинал, аналог или б/у
        </p>
      </div>
      <div className={styles.gift}>
        <img src={gloves} alt="" />
        <p className={styles.title}>
          <strong>Перчатки<br /> в подарок</strong>
        </p>
      </div>
      <div className={styles.gift}>
        <img src={chat} alt="" />
        <p className={styles.title}>
          <strong>Фото и видео запчастей</strong>
          <br /> пришлем в Whatsapp
        </p>
      </div>
    </div>
  </div>
);

export default QuizGifts;
