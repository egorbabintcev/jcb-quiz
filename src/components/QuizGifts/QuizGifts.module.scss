@import 'assets/sass/utils/functions';

.title {
  font: {
    size: rem(18);
    weight: 100;
  }
  margin: 0;
  margin-bottom: rem(20);
}

.gifts {
  display: flex;
  flex-direction: column;
  margin-top: auto;

  &_wrapper {
    display: flex;
    justify-content: space-between;
  }
}

.gift {
  align-items: center;
  display: flex;
  flex-direction: column;
  max-height: rem(110);

  .title {
    font: {
      size: rem(12);
      weight: 100;
    }
    line-height: 1.3;
    margin: 0;
    margin-bottom: auto;
    max-width: rem(120);
    text-align: center;

    strong {
      font: {
        size: rem(13);
        weight: 300;
      }
    }
  }

  img {
    margin-bottom: rem(8);
    max-height: rem(55);
    min-height: rem(55);
    // position: absolute;
  }
}
