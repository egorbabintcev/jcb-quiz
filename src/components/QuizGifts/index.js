import QuizGifts from './QuizGifts';

export default QuizGifts;
