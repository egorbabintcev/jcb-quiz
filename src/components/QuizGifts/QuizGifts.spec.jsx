import React from 'react';
import { render } from '@testing-library/react';
import QuizGifts from './QuizGifts';

describe('QuizGifts component', () => {
  it('Should render component with title and 3 gifts', () => {
    const { asFragment } = render(<QuizGifts />);
    expect(asFragment(<QuizGifts />)).toMatchSnapshot();
  });
})
