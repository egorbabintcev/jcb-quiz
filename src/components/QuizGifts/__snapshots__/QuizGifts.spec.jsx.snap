// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizGifts component Should render component with title and 3 gifts 1`] = `
<DocumentFragment>
  <div
    class="gifts"
  >
    <h4
      class="title"
    >
      Гарантированные подарки:
    </h4>
    <div
      class="gifts_wrapper"
    >
      <div
        class="gift"
      >
        <img
          alt=""
          src="prices.png"
        />
        <p
          class="title"
        >
          <strong>
            3 варианта цены
          </strong>
          <br />
           на оригинал, аналог или б/у
        </p>
      </div>
      <div
        class="gift"
      >
        <img
          alt=""
          src="gloves.png"
        />
        <p
          class="title"
        >
          <strong>
            Перчатки
            <br />
             в подарок
          </strong>
        </p>
      </div>
      <div
        class="gift"
      >
        <img
          alt=""
          src="chat.png"
        />
        <p
          class="title"
        >
          <strong>
            Фото и видео запчастей
          </strong>
          <br />
           пришлем в Whatsapp
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
